File: 30-Days-of-Code/Day-2-Operators/Rust/solution.rs
use std::io::{self,BufRead,Write};
use std::fmt;

fn main() {
    let mut line = String::new();
    let stdin = io::stdin();
    
    line.clear();    
    io::stdout().flush();
    stdin.lock().read_line(&mut line).unwrap();
    let meal_cost = line.trim().parse::<f32>().unwrap();
    
    line.clear(); 
    io::stdout().flush();
    stdin.lock().read_line(&mut line).unwrap();
    let tip_percent = line.trim().parse::<u32>().unwrap();
    
    line.clear(); 
    io::stdout().flush();
    stdin.lock().read_line(&mut line).unwrap();
    let tax_percent = line.trim().parse::<u32>().unwrap();
    
    let tip = meal_cost * (tip_percent as f32 / 100.0);
    let tax = meal_cost * (tax_percent as f32 / 100.0);
    let total_cost = meal_cost + tip + tax;
    
    let rounded_total_cost = total_cost.round();
    
    println!("The total meal cost is {:?} dollars.", rounded_total_cost);
}

File: 30-Days-of-Code/Day-2-Operators/Swift/solution.swift
import Foundation

var mealCost = Double(readLine()!)!
var tipPercent = Int(readLine()!)!
var taxPercent = Int(readLine()!)!

var tip = mealCost * (Double(tipPercent) / 100.0)
var tax = mealCost * (Double(taxPercent) / 100.0)
var totalCost = mealCost + tip + tax

var roundedTotalCost = Int(round(totalCost))

print("The total meal cost is \(roundedTotalCost) dollars.")
